### PR TITLE
Introduce the TO_ONE relationship used by the calculated field

### PR DESCRIPTION
**Pull request by goldmedal**
# Some follow-up issue
- Add more test case
- Test to link more relationship in one field

### CODE REVIEW

File: wren-modeling-rs/src/logical_plan/rule.rs [hunk 1]
```diff
@@ -199,13 +216,67 @@ impl ModelPlanNode {
...
+                        );
+                        expr.alias(column.name.clone())
+                    } else {
+                        col(column.name.clone())
+                    };
+                    required_exprs_buffer.insert(OrdExpr::new(expr_plan));
                 };
-                required_exprs_buffer.push_back(expr_plan);
 
+                if column.is_calculated {
```
Comment (grieve54706) [inline wren-modeling-rs/src/logical_plan/rule.rs:233]: it could be `else`.

File: wren-modeling-rs/src/logical_plan/rule.rs [hunk 2]
```diff
@@ -199,13 +216,67 @@ impl ModelPlanNode {
...
+                        let expr = mdl::utils::create_wren_calculated_field_expr(
+                            column_rf.clone(),
+                            Arc::clone(&analyzed_wren_mdl),
+                        );
+                        let expr_plan = expr.alias(column.name.clone());
+                        required_exprs_buffer.insert(OrdExpr::new(expr_plan));
+                    };
+
+                    let qualified_column =
+                        Column::from_qualified_name(format!("{}.{}", model.name, column.name));
```
Comment (grieve54706) [inline wren-modeling-rs/src/logical_plan/rule.rs:249]: How about adding a util func to format qualified name?

File: wren-modeling-rs/src/logical_plan/rule.rs [hunk 3]
```diff
@@ -217,19 +288,204 @@ impl ModelPlanNode {
...
+                                ),
+                                link.join_type,
+                                link.condition.clone(),
+                                Box::new(relation_chain),
+                            );
+                        }
+                        _ => {
+                            unimplemented!("Only support model as source dataset")
+                        }
+                    }
```
Comment (grieve54706) [inline wren-modeling-rs/src/logical_plan/rule.rs:332]: If only one variants need to process. How about using if-else?

File: wren-modeling-rs/src/mdl/lineage.rs [hunk 4]
```diff
@@ -175,21 +177,68 @@ impl Lineage {
...
+) -> DatasetLink {
+    let join_type = if rs.models[0] == source.get_name() {
+        rs.join_type
+    } else {
+        match rs.join_type {
+            JoinType::OneToOne => JoinType::OneToOne,
+            JoinType::OneToMany => JoinType::ManyToOne,
+            JoinType::ManyToOne => JoinType::OneToMany,
+            JoinType::ManyToMany => JoinType::ManyToMany,
+        }
```
Comment (grieve54706) [inline wren-modeling-rs/src/mdl/lineage.rs:226]: How about using`_` to accentuate changed variants?
```rust
match rs.join_type {
    JoinType::OneToMany => JoinType::ManyToOne,
    JoinType::ManyToOne => JoinType::OneToMany,
    (_) => rs.join_type
}
```

File: wren-modeling-rs/src/mdl/lineage.rs [hunk 5]
```diff
@@ -247,44 +296,76 @@ mod test {
...
-            .requried_fields_map
+        let customer_name = lineage
+            .required_fields_map
             .get(&Column::from_qualified_name("orders.customer_name"))
-            .unwrap()
+            .unwrap();
+        let mut expected = HashSet::new();
+        expected.insert(Column::from_qualified_name("customer.name"));
+        expected.insert(Column::from_qualified_name("orders.custkey"));
+        expected.insert(Column::from_qualified_name("customer.custkey"));
```
Comment (grieve54706) [inline wren-modeling-rs/src/mdl/lineage.rs:332]: How about using `HashSet::from`?

File: wren-modeling-rs/src/mdl/mod.rs [hunk 6]
```diff
@@ -118,15 +134,19 @@ pub fn transform_sql(wren_mdl: Arc<WrenMDL>, sql: &str) -> Result<String, DataFu
...
 
     let config = ConfigOptions::default();
 
-    let analyzed = analyzer
-        .execute_and_check(plan, &config, |_, _| {})
-        .unwrap();
+    let analyzed = match analyzer.execute_and_check(plan, &config, |_, _| {}) {
+        Ok(analyzed) => analyzed,
+        Err(e) => {
+            println!("Error: {:?}", e);
```
[the remark this answers is not in the document]
Comment (goldmedal) [reply wren-modeling-rs/src/mdl/mod.rs:146]: Maybe use `eprintln` instead. I saw DataFusion prefers to use both println or eprintln.

File: wren-modeling-rs/src/mdl/utils.rs [hunk 7]
```diff
@@ -42,3 +55,187 @@ where
...
+                &projection.input,
+            )
+            .unwrap_or_else(|_| panic!("Failed to create column expression {}", expr))
+        }
+        _ => unreachable!("Unexpected plan type: {:?}", plan),
+    }
+}
+
+/// Create the Logical Expr for the remote column.
+/// The
```
[the remark this answers is not in the document]
Comment (goldmedal) [reply wren-modeling-rs/src/mdl/utils.rs:153]: My mistake. I'll complete it.

**Review by grieve54706 — APPROVED** [review]
LGTM

**Pull request merged**